common/src/escrow_accounts.rs

```diff
@@ -99,27 +98,17 @@ type BigInt = String;
 )]
 pub struct EscrowAccountQuery;
 
-pub fn escrow_accounts(
+pub async fn escrow_accounts(
     escrow_subgraph: &'static SubgraphClient,
     indexer_address: Address,
     interval: Duration,
     reject_thawing_signers: bool,
-) -> Eventual<EscrowAccounts> {
-    timer(interval).map_with_retry(
-        move |_| async move {
-            get_escrow_accounts(escrow_subgraph, indexer_address, reject_thawing_signers)
-                .await
-                .map_err(|e| e.to_string())
-        },
-        move |err: String| {
-            error!(
-                "Failed to fetch escrow accounts for indexer {:?}: {}",
-                indexer_address, err
-            );
-
-            sleep(interval.div_f32(2.0))
-        },
-    )
+) -> Receiver<EscrowAccounts> {
+    watcher::new_watcher(interval, move || {
+        get_escrow_accounts(escrow_subgraph, indexer_address, reject_thawing_signers)
+    })
+    .await
+    .unwrap()
```

Comment:
could you return this as a Result and do an expect (or unwrap in case of tests) where this is called?

Reply:
hi, you meant to change the return type right?